tests/system/test_identifier.py

```diff
@@ -0,0 +1,148 @@
+import pandas as pd
+import numpy as np
+from csodiaq.identifier import Identifier
+from csodiaq.loaders.query import QueryLoaderContext
+import os
+import pickle
+import pytest
+
+
+def get_parent_dir():
+    return os.path.dirname(os.path.abspath(__file__))
+
+
+def get_file_from_system_test_folder(file):
+    return os.path.join(get_parent_dir(), "test_files", file)
+
+
+@pytest.fixture
+def commandLineArgs():
+    args = {
+        "command": "id",
+        "files": [get_file_from_system_test_folder("20190412_DI2A_1to1_tMS2_n3.mzXML")],
+        "library": get_file_from_system_test_folder("system_test_library.csv"),
+        "outDirectory": "",
+        "fragmentMassTolerance": 30,
+        "correction": 0,
+        "histogram": False,
+        "proteinTargets": 1,
+        "query": 0,
+        "heavyMz": True,
+        "peaks": 0,
+        "commonpeptide": False,
+        "commonprotein": False,
+    }
+    return args
+
+
+def assert_numeric_pandas_dataframes_are_equal(expectedDf, df, type):
+    assert len(expectedDf.index) == len(df.index)
+    columns = get_columns_that_should_match(type)
+    for columnName in columns:
+        assert columnName in df.columns
+        expectedColumn = np.array(expectedDf[columnName])
+        column = np.array(df[columnName])
+        expectedColumn.sort()
+        column.sort()
+        if expectedColumn.dtype.kind in np.typecodes["AllFloat"]:
+            np.testing.assert_array_almost_equal(expectedColumn, column)
+        else:
+            np.testing.assert_array_equal(expectedColumn, column)
```

Comment:
```Python
assert expectedDf.equals(df)
```

Reply:
I made the custom assert function for comparing dataframes because some of the outputs from the main branch differ slightly from the outputs of the current dev branch. For example, I've included an "isDecoy" column in the output of the dev branch to simplify some of the FDR calculations. Because these initial system tests are comparing it to the output of the old program, I've selected the "important" columns and compared those. My thoughts are that I will complete this initial rewrite verifying that the dev output references the main output, then after that I will reset the test files to reference the dev output at that time, making direct dataframe comparisons possible.

Reply:
On that note, though, I can clean this function up a bit. I just pushed some changes regarding how I sort the comparisons.